### PR TITLE
Return Environment on stop.

### PR DESCRIPTION
**Pull request by hsmade**
This removes manual parsing of the response json in clients, when they want to know the state, for example.

### CODE REVIEW

**Review by timofurrer — APPROVED** [review]
Awesome, thanks!

File: environments.go [hunk 1]
```diff
@@ -208,17 +208,19 @@ func (s *EnvironmentsService) DeleteEnvironment(pid interface{}, environment int
...
 	req, err := s.client.NewRequest(http.MethodPost, u, nil, options)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return s.client.Do(req, nil)
+	env := new(Environment)
+	resp, err := s.client.Do(req, env)
+	return env, resp, err
```
Comment (svanharmelen) [inline environments.go:225]:
```suggestion
	env := new(Environment)
	resp, err := s.client.Do(req, env)
	if err != nil {
		return nil, resp, err
	}
	
	return env, resp, nil
```
For consistency... Also noticed this in this file the returns always return `err` even when the `err` is already checked. So feel free (please do 😉 ) to update the other return statements as well 😊

**Review by svanharmelen — CHANGES_REQUESTED** [review]

File: environments.go (hunk 1 above)
Comment (hsmade) [reply environments.go]: Hmm not only this file I see. 2 other random files I checked also always return err, even when it's already checked. 
Want me to do a search replace in a new PR maybe?

Comment (svanharmelen) [reply environments.go]: Sure... I prefer to keep things consistent, so please do 😊

Comment (hsmade) [reply environments.go]: https://github.com/xanzy/go-gitlab/pull/1710

**Review by svanharmelen — APPROVED** [review]
Thanks, LGTM 👍🏻

**Comment by svanharmelen** [comment]
@hsmade can you check and fix the test? When it passes I'll merge it...

**Comment by hsmade** [comment]
Took me a bit of debugging. Which is fun because it just works when you run it using delve. But in the end the fix makes sense actually.  It now returns an object, so it needs some json from the API.

**Pull request merged**